src/nvim/normal.c

```diff
@@ -4388,10 +4386,10 @@ static void nv_ident(cmdarg_T *cap)
       STRCAT(buf, "\\>");
     /* put pattern in search history */
     init_history();
-    add_to_history(HIST_SEARCH, buf, true, NUL);
-    normal_search(cap, cmdchar == '*' ? '/' : '?', buf, 0);
+    add_to_history(HIST_SEARCH, (char_u *)buf, true, NUL);
```

Comment:
I see casts from `(char_u *)` to `(char *)` everywhere, but isn't that a narrowing conversion? Is it safe to cast from `(char *)` to `(char_u *)` (widening conversion)?